### ISSUE TITLE
Cake Environment does not detect running on OSX 

### ISSUE THREAD

**Issue by Julien-Mialon** [cake-build/cake#1407]
### What You Are Seeing?
CakeContext.Environment.Platform.Family return Linux when running on OSX.

### What is Expected?
PlatformFamily.OSX

### What version of Cake are you using?
0.17

### Are you running on a 32 or 64 bit system?
64bits

### What environment are you running on?  Windows? Linux? Mac?
Mac OS X El capitan

### Are you running on a CI Server?  If so, which one?
No

### How Did You Get This To Happen? (Steps to Reproduce)
Create an addins using CakeContext.Environment.Platform.Family to check if running on OS X
```
[CakeMethodAlias]
public static void LogPlatform(this ICakeContext context)
{
    context.Log.Write(Verbosity.Normal, LogLevel.Information, $"Family: {context.Environment.Platform.Family}");
}

```

**Comment by agc93** [comment]
Just to be clear (since they use different implementations), this is using "normal"/Mono-run Cake, rather than Cake.CoreCLR, yes?

**Comment by Julien-Mialon** [comment]
Yes exactly, running on mono implementation

**Comment by agc93** [comment]
Okay, I would imagine this is caused by Mono returning an ID of 4 for macOS/OS X for legacy/compat reasons. The Mono FAQ mentions that [here](http://www.mono-project.com/docs/faq/technical/#how-to-detect-the-execution-platform).

@patriksvensson Any thoughts here? Obviously the .NET Core implementation can use the very neat `RuntimeInformation` bits, and changing to Roslyn will also fix this, but any thoughts on making this work for Mono, annoying legacy behaviour and all?

**Comment by patriksvensson** [comment]
@agc93 The .NET Core implementation use `RuntimeInformation`. We need to fix the non .NET Core implementation somehow.

**Comment by patriksvensson** [comment]
@agc93 This is what fails: https://github.com/cake-build/cake/blob/develop/src/Cake.Core/Polyfill/EnvironmentHelper.cs#L46

**Comment by agc93** [comment]
Yeah, that's what I was looking at: Mono returns a 4 for compat/legacy reasons. Any way to fix from our end you can think of?

**Comment by patriksvensson** [comment]
@agc93 Not at the top of my head. Will investigate when I have some time. Can't test this though since I don't have a Mac. Perhaps @Julien-Mialon can try it out for us?

**Comment by Julien-Mialon** [comment]
@patriksvensson I just tested it with this small program : 
```csharp
var platform = Environment.OSVersion.Platform;
Console.WriteLine("platform={0}", platform);
Console.WriteLine("int={0}", (int)platform);
```
The result is : 
```bash
platform=Unix
int=4
```

Hope that can help you.

**Comment by augustoproiete** [comment]
This has been resolved since Cake v0.30.0.

```csharp
Information("Environment.Platform.Family: {0}", Context.Environment.Platform.Family);
```

Results in:

```
Environment.Platform.Family: OSX
```

Cake v1.0.0 introduces a number of [helper methods including `IsRunningOnMacOs`](https://cakebuild.net/api/Cake.Common/EnvironmentAliases/#Methods), which returns `true`/`false`.

```csharp
Information("IsRunningOnMacOs: {0}", Context.IsRunningOnMacOs());
Information("Environment.Platform.Family: {0}", Context.Environment.Platform.Family);
```

Results in:

```
IsRunningOnMacOs: True
Environment.Platform.Family: OSX
```

---

[![image](https://user-images.githubusercontent.com/177608/106851655-0679ce80-668d-11eb-8741-af5aa89f85c6.png)](https://cakebuild.net/api/Cake.Common/EnvironmentAliases/#Methods)
